e2e/tests-dfx/assetscanister.bash

```diff
@@ -506,3 +506,45 @@ CHERRIES" "$stdout"
     assert_match "HTTP/1.1 404 Not Found"
 
 }
+
+@test "asset configuration via .ic-assets.json5 - overwriting default headers" {
+    install_asset assetscanister
+
+    dfx_start
+
+    touch src/e2e_project_frontend/assets/thing.json
+    touch src/e2e_project_frontend/assets/.ignored-by-defualt.txt
+
+    mkdir src/e2e_project_frontend/assets/.well-known
+    touch src/e2e_project_frontend/assets/.well-known/thing.json
+
+    mkdir src/e2e_project_frontend/assets/.well-known/.hidden
+    touch src/e2e_project_frontend/assets/.well-known/.hidden/ignored.txt
+
+    mkdir src/e2e_project_frontend/assets/.well-known/.another-hidden
+    touch src/e2e_project_frontend/assets/.well-known/.another-hidden/ignored.txt
+
+    echo '[
+      {
+        "match": "thing.json",
+        "cache": { "max_age": 2000 },
+        "headers": {
+          "Content-Encoding": "my-encoding",
+          "Content-Type": "x-type",
+          "Cache-Control": "custom",
+          "etag": "my-etag"
+        }
+      }
+    ]' > src/e2e_project_frontend/assets/.ic-assets.json5
+
+    dfx deploy
+
+    ID=$(dfx canister id e2e_project_frontend)
+    PORT=$(get_webserver_port)
+
+    assert_command curl --head "http://localhost:$PORT/thing.json?canisterId=$ID"
+    assert_match "cache-control: custom"
+    assert_match "content-encoding: my-encoding"
+    assert_match "content-type: x-type"
+    assert_not_match "etag: my-tag"
```

Comment:
i think it doesn't (uses `assert_not_match`)

Reply:
Yes, that's the point.  If I understand it, this `assert_not_match "etag: my-tag"` is meant to demonstrate that the `"etag": ...` line from the configuration does not take effect.

However, since `my-tag` != `my-etag`, it doesn't prove anything.  The string `my-tag` does not appear in the configuration.

I'll ask another way.

Suppose this line read `assert_not_match "etag: elephant"`. What would this assertion prove?

Next, what does `assert_not_match "etag: my-tag"` prove?

Reply:
dang it, I completely missed `my-tag` != `my-etag`

Reply:
fixed with
```
    assert_not_match "etag: my-etag"
    assert_match "etag: \"[a-z0-9]{64}\""
```

thanks Eric!